src/lib/createStardustComponent.tsx

```diff
@@ -0,0 +1,46 @@
+import createComponentInternal from './createComponent'
+import * as React from 'react'
+import * as _ from 'lodash'
+import {
+  ComponentSlotClasses,
+  ThemePrepared,
+  ComponentSlotStylesPrepared,
+  ComponentVariablesObject,
+} from '../themes/types'
+
+export interface RenderStardustResultConfig {
+  classes: ComponentSlotClasses
+  variables: ComponentVariablesObject
+  styles: ComponentSlotStylesPrepared
+  rtl: boolean
+  theme: ThemePrepared
+}
+
+export type RenderStardustComponentCallback<P> = (
+  config: RenderStardustResultConfig,
+  props: P,
+) => any
+
+export interface CreateStardustComponentConfig<P> {
+  displayName?: string
+  render: RenderStardustComponentCallback<P>
+}
+
+const createComponent = <P extends {} = {}, S extends {} = {}>({
+  displayName = 'StardustComponent',
+  render,
+}: CreateStardustComponentConfig<P>): React.SFC<P> => {
+  const StardustComponent: React.SFC<P> = createComponentInternal({
+    displayName,
```

Comment:
is there any reason we are using this value both here, as well as on the following line: https://github.com/stardust-ui/react/pull/503/files#diff-3cd0aeac45681cbb710926892d4cf28bR42 ? Maybe I am missing something, just would expect that there will be the only place where this is used

Reply:
We don't need it on the second usage, just leftover from the renderComponent. Will remove it.